cartography/intel/gcp/__init__.py

```diff
@@ -1,12 +1,102 @@
+
+
+def _get_crm_resource_v2(credentials):
+    """
+    Instantiates a Google Compute Resource Manager v2 resource object to call the Resource Manager API.
+    We need a v2 resource object to query for GCP folders.
+    :param credentials: The GoogleCredentials object
+    :return: A CRM v2 resource object
+    """
+    return googleapiclient.discovery.build('cloudresourcemanager', 'v2', credentials=credentials, cache_discovery=False)
+
+
+def _get_compute_resource(credentials):
+    """
+    Instantiates a Google Compute resource object to call the Compute API. This is used to pull zone, instance, and
+    networking data. See https://cloud.google.com/compute/docs/reference/rest/v1/.
+    :param credentials: The GoogleCredentials object
+    :return: A Compute resource object
+    """
+    return googleapiclient.discovery.build('compute', 'v1', credentials=credentials, cache_discovery=False)
+
+
+def _initialize_resources(credentials):
+    """
+    Create namedtuple of all resource objects necessary for GCP data gathering.
+    :param credentials: The GoogleCredentials object
+    :return: namedtuple of all resource objects
+    """
+    return Resources(
+        crm_v1=_get_crm_resource_v1(credentials),
+        crm_v2=_get_crm_resource_v2(credentials),
+        compute=_get_compute_resource(credentials)
+    )
+
+
+def _sync_single_project(session, resources, project_id, gcp_update_tag, common_job_parameters):
+    """
+    Handles graph sync for a single GCP project.
+    :param session: The Neo4j session
+    :param resources: namedtuple of the GCP resource objects
+    :param project_id: The project ID number to sync.  See  the `projectId` field in
+    https://cloud.google.com/resource-manager/reference/rest/v1/projects
+    :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
+    :param common_job_parameters: Other parameters sent to Neo4j
+    :return: Nothing
+    """
+    compute.sync_gcp_instances(session, resources.compute, project_id, gcp_update_tag, common_job_parameters)
+
+
+def _sync_multiple_projects(session, resources, projects, gcp_update_tag, common_job_parameters):
+    """
+    Handles graph sync for multiple GCP projects.
+    :param session: The Neo4j session
+    :param resources: namedtuple of the GCP resource objects
+    :param: projects: A list of projects. At minimum, this list should contain a list of dicts with the key "projectId"
+     defined; so it would look like this: [{"projectId": "my-project-id-12345"}].
+    This is the returned data from `crm.get_gcp_projects()`.
+    See https://cloud.google.com/resource-manager/reference/rest/v1/projects.
+    :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
+    :param common_job_parameters: Other parameters sent to Neo4j
+    :return: Nothing
+    """
+    crm.sync_gcp_projects(session, projects, gcp_update_tag, common_job_parameters)
+
+    for project in projects:
+        project_id = project['projectId']
```

Comment:
See AWS module: log an INFO-level message that this GCP project is being synced.

https://github.com/lyft/cartography/blob/master/cartography/intel/aws/__init__.py#L59